src/client/Microsoft.Identity.Client/ManagedIdentityApplication.cs

```diff
@@ -55,13 +55,28 @@ public AcquireTokenForManagedIdentityParameterBuilder AcquireTokenForManagedIden
                 resource);
         }
 
+        /// <inheritdoc/>
+        public async Task<ManagedIdentitySource> GetManagedIdentitySourceAsync()
```

Comment:
Why not keep this as static as well. The method was created static so that Azure Identity can detect if managed identity is available before creating the source. Was this discussed with them?

Reply:
*sigh* this is the first time I've heard that. CC @bgavrilMS what are your thoughts on this?

Reply:
@gladjohn Can you confirm that this would work with Azure Identity? Did you discuss with them?

Reply:
The dev experience won't change for Azure SDK. You, Bogdan and I can discuss over chat.